pkg/querier/queryrange/results_cache.go

```diff
@@ -72,6 +74,12 @@ type resultsCache struct {
 }
 
 // NewResultsCacheMiddleware creates results cache middleware from config.
+// The middleware cache result using a unique cache key for a given request (step,query,user) and interval.
+// The cache assumes that each request length (end-start) is below or equal the interval.
+// Each request starting from within the same interval will hit the same cache entry, if
+// the entry extents (array of responses already made) doesn't contains enough data it will query the next
```

Comment:
If the cache doesn't have the entire duration of the request cached, it will query the uncached parts and append them to the cache entries.

Reply:
yes that's definitely clearer